### ISSUE TITLE
Sector sizes above 4KB are unsupported.

### ISSUE THREAD

**Issue by PoeticDeath** [winfsp/winfsp#489]
I have not yet looked into the page size issue, I intend to. However I did some benchmarking with three different versions of WinFsp. These versions being the latest stable release, the latest commit to master, and my own fork. Currently my newest build 3.4 passes all 67 tests and supports sector sizes upto 2GB. All benchmarks and tests were run on 64MB sector sizes on [SpaceFS V1.5.3](https://github.com/PoeticDeath/SpaceFS) running on top of WinFsp. WinFsp autmatically tells Windows to split into 4KB chunks for sector sizes above 4KB (Or 2GB chunks for sector sizes above 2GB for 3.4). This is done using:  
`Info->LogicalBytesPerSector =  
Info->PhysicalBytesPerSectorForAtomicity =  
Info->PhysicalBytesPerSectorForPerformance =  
Info->FileSystemEffectivePhysicalBytesPerSectorForAtomicity =  
FsvolDeviceExtension->VolumeParams.SectorSize;`  
in volinfo.c lines 206-210. The speed difference between master and 3.4 is minimal, however on the stable and master builds, 11 WinFsp tests fail. All with the same error: `ASSERT(BytesPerSector == BytesTransferred)` These same tests pass under or at a 4KB sector size. In 3.4 they pass with any valid sector size (Powers of 2, 512 - 2147483648).  
Stable:  
![](https://user-images.githubusercontent.com/46275713/212193877-bab6c0cb-b34d-44d2-8ee4-e62e67fe8e55.jpg)
Master:  
![](https://user-images.githubusercontent.com/46275713/212193892-5f23b26b-146e-4662-97a2-7647fbb8b3ed.jpg)
3.4:  
![](https://user-images.githubusercontent.com/46275713/212193905-41f9c00e-eae5-46cc-8f00-d86b0c665085.jpg)

**Comment by PoeticDeath** [comment]
Updated to 3.5 should operate the same but the old SectorSize value is now Reserved16 and Reserved32 is now SectorSize, This required updating a test. Also added contributor info. All 67 tests pass successfully, the failures on my branch are due to time restrictions of AppVeyor.

I looked quite a bit into the page size... I have found according to a [Microsoft DevBlog](https://devblogs.microsoft.com/oldnewthing/20210510-00/?p=105200), large page sizes of 2MB are also supported on all architectures we have with WinFsp that being x86 PAE, x64, and ARM64... unless I missed one? So doing my due diligence I looked for all hardcodes of '4096' being the old page size and some key words like 'page' and 'alloc' but nothing seemed to pop out like it did with the sector size. I also looked for the [VirtualAlloc](https://learn.microsoft.com/en-us/windows/win32/api/memoryapi/nf-memoryapi-virtualalloc) function mentioned in #488 but this did not yield any results.
